test/e2e/util/e2e_client.go

```diff
@@ -59,6 +59,11 @@ func (m *E2EKubeClient) Reset() error {
 			break
 		}
 
+		// Allow resources to opt-out of cleanup at the end of the test.
```

Comment:
```suggestion
		// Allow resources to opt-out of being deleted by the e2e client.
```

Reply:
In my recent work on e2es I've added `$SKIP_CLEANUP` checks to stop cleaning up - would that approach work here?

Reply:
Good idea! If we decide to split this out as proposed [here](https://github.com/operator-framework/operator-lifecycle-manager/pull/3042#discussion_r1337494471) I'll share this idea in the ticket.